Assembler.UnitTests/AssemblerTests.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+using Moq;
+using NUnit.Framework;
+
+namespace Assembler.UnitTests
+{
+    [TestFixture]
+    public class MessagesAssemblerTests
+    {
+        private Mock<IResolver<FrameType, IHandler<BaseFrame, BaseMessageInAssembly>>> _resolverMock;
+        private Mock<IConverter<BaseMessageInAssembly, BaseAssembledMessage>> _converterMock;
+        private Mock<ITimeBasedCache<BaseMessageInAssembly>> _cacheMock;
+        private Mock<IHandler<BaseFrame, BaseMessageInAssembly>> _firstHandlerMock;
+        private Mock<IHandler<BaseFrame, BaseMessageInAssembly>> _secondHandlerMock;
+        private Base.MessagesAssembler _assembler;
+
+        private List<BaseAssembledMessage> _assembledMessages;
+        private List<Mock<IHandler<BaseFrame, BaseMessageInAssembly>>> _handlersList;
+
+        [SetUp]
+        public void Setup()
+        {
+            _assembledMessages = new List<BaseAssembledMessage>();
+            _resolverMock = new Mock<IResolver<FrameType, IHandler<BaseFrame, BaseMessageInAssembly>>>();
+            _converterMock = new Mock<IConverter<BaseMessageInAssembly, BaseAssembledMessage>>();
+            _cacheMock = new Mock<ITimeBasedCache<BaseMessageInAssembly>>();
+
+            _firstHandlerMock = new Mock<IHandler<BaseFrame, BaseMessageInAssembly>>();
+            _secondHandlerMock = new Mock<IHandler<BaseFrame, BaseMessageInAssembly>>();
+
+            _handlersList = new List<Mock<IHandler<BaseFrame, BaseMessageInAssembly>>>
+            {
+                _firstHandlerMock,
+                _secondHandlerMock
+            };
+            var handlersList = _handlersList.Select(handlers => handlers.Object);
```

Comment:
Define the list as List<IHandler<BaseFrame, BaseMessageInAssembly>>
and use _firstHandlerMock.Object

Reply:
But the class gets a list